packages/astro/src/vite-plugin-env/index.ts

```diff
@@ -68,6 +71,13 @@ export default function envVitePlugin({ config: astroConfig }: EnvPluginOptions)
 				if (privateEnv) {
 					const entries = Object.entries(privateEnv).map(([key, value]) => [`import.meta.env.${key}`, value]);
 					replacements = Object.fromEntries(entries);
+					// These additional replacements are needed to match Vite
+					replacements = Object.assign(replacements, {
+						// This catches destructed `import.meta.env` calls,
+						// BUT we only want to inject private keys referenced in the file.
+						// We overwrite this value on a per-file basis.
+						'import.meta.env': `({})`,
```

Comment:
This was the main fix, we have to replace `import.meta.env` _in addtion to_ `import.meta.env.NAME`